docs/elements/guides/styling.mdx

```diff
@@ -0,0 +1,270 @@
+})
+```
+
+</Callout>
+
+### CSS Modules
+
+Classes from an imported CSS module can be applied to most Clerk Elements components.
+
+```tsx filename="/app/sign-in/[[...sign-in]]/page.tsx"
+import * as Clerk from '@clerk/elements/common'
+import * as SignIn from '@clerk/elements/sign-in'
+import styles from './sign-in.module.css'
+
+export default function SignInPage() {
+  return (
+    <SignIn.Root>
+      <SignIn.Step name="start" className={styles.startStep}>
+        <Clerk.Connection name="google" className={styles.provider}>Sign in with Google</Clerk.Connection>
+        <Clerk.Field name="identifier">
+          <Clerk.Label className={styles.label}>Email</Clerk.Label>
+          <Clerk.Input className={styles.input} />
+          <Clerk.FieldError className={styles.error} />
+        </Clerk.Field>
+        <SignIn.Action submit className={styles.submit}>Continue</SignIn.Action>
+      </SignIn.Step>
+      <SignIn.Step name="verifications" className={styles.verificationsStep}>
+        <SignIn.Strategy name="email_code">
+          <Clerk.Field name="code">
+            <Clerk.Label className={styles.label}>Code</Clerk.Label>
+            <Clerk.Input className={styles.input} />
+            <Clerk.FieldError className={styles.error} />
+          </Clerk.Field>
+          <SignIn.Action submit className={styles.submit}>Verify</SignIn.Action>
+      </SignIn.Step>
+    </SignIn.Root>
+  )
+}
+```
+
+### Inline styles
+
+Just like native HTML elements, you can also use inline styles to style Clerk Elements. This is useful when you need to apply styles conditionally or when you want to avoid creating a separate CSS file.
+
+```tsx filename="/app/sign-in/[[...sign-in]]/page.tsx"
+import * as Clerk from '@clerk/elements/common'
+import * as SignIn from '@clerk/elements/sign-in'
+
+export default function SignInPage() {
+  return (
+    <SignIn.Root>
+      <SignIn.Step name="start" style={{
+        backgroundColor: 'white',
+        padding: '1rem',
+        border: '1px solid #dddddd',
+        borderRadius: '4px',
+      }}>
+        <Clerk.Field name="identifier">
+          <Clerk.Label style={{
+            color: '#dddddd',
+            fontSize: '0.875rem',
+          }}>Email</Clerk.Label>
+          <Clerk.Input style={{
+            border: '1px solid #dddddd',
+            borderRadius: '4px',
+            padding: '1rem',
+          }} />
+        </Clerk.Field>
+        <SignIn.Action submit style={{
+          backgroundColor: '#111111',
+          color: 'white',
+          padding: '1rem',
+          borderRadius: '4px',
+        }}>Continue</SignIn.Action>
+      </SignIn.Step>
+    </SignIn.Root>
+  )
+}
+```
+
+## State-based styling
+
+In some cases you might want to style components based on their state. Clerk Elements exposes data attributes for this purpose, as well as components that expose the state programmatically to support more complex logic.
+
+### Data attributes
+
+#### Field components
```

Comment:
```suggestion
```
I think this heading isn't totally necessary, and removing it helps us avoid an empty heading